File: FiveCardStud/julia/Card.jl
# #!/usr2/local/julia-1.10.4/bin/julia
include("CommonData.jl")

#suit is first here but not in constructor so they do not overload each other
mutable struct Card
	suit::Char
	value::Int
end

function Card( newValue::Int, newSuit::Char )
	return Card(newSuit, newValue)
end

function suitRank(card::Card)
	if card.suit == 'D'
		return 1
	elseif card.suit == 'C'
		return 2
	elseif card.suit == 'H'
		return 3
	elseif card.suit == 'S'
		return 4
	else
		return -1
	end
end

function getValue(c::Char)
	i = Int(c)
	if i >= Int('2') && i <= Int('9')
		return (parse(Int, c)-1)
	elseif c == 'J'
		return 10
	elseif c == 'Q'
		return 11
	elseif c == 'K'
		return 12
	elseif c == 'A'
		return 13
	else
		return -1
	end
end

function toString(card::Card)
	return FACES[card.value] * card.suit
end

File: FiveCardStud/rust/src/card.rs
#[derive(Copy, Clone)]
pub struct Card {
    pub value: i32,
    pub suit: char,
}

impl Card {
    pub fn new(value: i32, suit: char) -> Self {
        Self { value, suit }
    }

    pub fn default() -> Self{
        Card { value : 0, suit: 'z' }
    }

    pub fn suit_rank(&self) -> i32 {
        match self.suit {
            'D' => 0,
            'C' => 1,
            'H' => 2,
            'S' => 3,
            _ => -1,
        }
    }

    pub fn value_from_char(c: char) -> i32 {
        match c {
            '2' => 0,
            '3' => 1,
            '4' => 2,
            '5' => 3,
            '6' => 4,
            '7' => 5,
            '8' => 6,
            '9' => 7,
            'J' => 9,
            'Q' => 10,
            'K' => 11,
            'A' => 12,
            _ => -1,
        }
    }

    pub fn to_string(&self) -> String {
        let faces = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K", "A"];
        format!("{}{}", faces[self.value as usize], self.suit)
    }
}